### PR TITLE
feat: Active state update on facade #8976

### LINKED ISSUE #8976: Active state update on facade
Most of the entities in my company can be disabled / enabled. While we have a method to patch the entity, there's not way to receive an update on wether the item was activated or deactivated. 

The UI has introduced a specific update for enabling/disabling the status, but it's done in an ugly way. We need to come up with precise updates instead.

### PR DESCRIPTION
**Pull request by Pucek9**
Closes #8976

### CODE REVIEW

File: feature-libs/my-account/organization/core/model/LoadStatus.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+export enum LoadStatus {
```
Comment (tobi-or-not-tobi) [inline feature-libs/my-account/organization/core/model/LoadStatus.ts]: this file doesn't follow naming convention; lower case, ending with `.model.ts`

File: feature-libs/my-account/organization/core/model/LoadStatus.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+//TODO: consider to make public LoaderState from core
+interface LoaderState<T> {
+  loading?: boolean;
+  error?: boolean;
+  success?: boolean;
+  value?: T;
+}
+
+export function mapToItemInfo<T>(currentState: LoaderState<T>): ItemInfo<T> {
```
Comment (tobi-or-not-tobi) [inline feature-libs/my-account/organization/core/model/LoadStatus.ts]: This is not a model, consider moving to another utility file.

File: feature-libs/my-account/organization/core/model/LoadStatus.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+export enum LoadStatus {
+  SUCCESS,
+  ERROR,
+}
+
+export interface ItemInfo<T> {
+  status: LoadStatus;
+  value: T;
```
Comment (tobi-or-not-tobi) [inline feature-libs/my-account/organization/core/model/LoadStatus.ts]: what about renaming to `item`? We've used item in a lot of places, and this one is even called ItemInfo.

File: feature-libs/my-account/organization/core/model/LoadStatus.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+export enum LoadStatus {
+  SUCCESS,
+  ERROR,
+}
+
+export interface ItemInfo<T> {
```
Comment (tobi-or-not-tobi) [inline feature-libs/my-account/organization/core/model/LoadStatus.ts]: what about `ItemStatus`?

File: feature-libs/my-account/organization/core/services/budget.service.spec.ts [hunk 5]
```diff
@@ -156,7 +157,7 @@ describe('BudgetService', () => {
   });
 
   describe('update budget', () => {
-    it('update() should should dispatch UpdateBudget action', () => {
+    it('update() should should be able to get status change from loading with value', () => {
```
Comment (Pucek9) [inline feature-libs/my-account/organization/core/services/budget.service.spec.ts]:
```suggestion
    it('update() should should dispatch UpdateBudget action', () => {
```

File: feature-libs/my-account/organization/core/model/LoadStatus.ts (hunk 4 above)
Comment (tobi-or-not-tobi) [reply feature-libs/my-account/organization/core/model/LoadStatus.ts]: we might also want to prefix those models with `Organization`, i.e. `OrganizationItemStatus `, to make them more specific.

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]
Only a few notes on the models. I've tested it with toggle status, works fine. 
I'll ask Stan to have a look at it as well.

File: feature-libs/my-account/organization/core/model/LoadStatus.ts (hunk 2 above)
Comment (marlass) [inline feature-libs/my-account/organization/core/model/LoadStatus.ts]: In the same util file for this method you can also crate util that takes the stream and does the `pairwise`, login check and then uses this function in map so it is not repeated in every service.

**Review by marlass — CHANGES_REQUESTED** [review]

**Comment by Pucek9** [comment]
QA steps:
you can test toggling by change toggle-status.component
```
protected update(item: T): void {
  this.itemService
    .update(item[this.key], this.getPatchedItem(item))
    .pipe(take(1))
    .subscribe(console.log);
}
```

**Review by tobi-or-not-tobi — DISMISSED** [review]

File: feature-libs/my-account/organization/components/unit/links/users/services/unit-user-item.service.ts [hunk 6]
```diff
@@ -26,7 +27,12 @@ export class UnitUserItemService extends OrganizationItemService<B2BUser> {
     return this.b2bUserService.get(unitUid);
   }
 
-  update(_customerId: string, _user: B2BUser) {}
+  update(
+    _customerId: string,
```
Comment (marlass) [inline feature-libs/my-account/organization/components/unit/links/users/services/unit-user-item.service.ts]: Should it also be prefixed with `_` as this variable is now used?

File: feature-libs/my-account/organization/components/unit/links/users/services/unit-user-item.service.ts [hunk 7]
```diff
@@ -26,7 +27,12 @@ export class UnitUserItemService extends OrganizationItemService<B2BUser> {
...
+    _user: B2BUser
+  ): Observable<OrganizationItemStatus<B2BUser>> {
+    return this.b2bUserService.getLoadingStatus(_customerId);
```
Comment (Pucek9) [inline feature-libs/my-account/organization/components/unit/links/users/services/unit-user-item.service.ts]:
```suggestion
    customerId: string,
    _user: B2BUser
  ): Observable<OrganizationItemStatus<B2BUser>> {
    return this.b2bUserService.getLoadingStatus(customerId);
```

**Review by marlass — APPROVED** [review]

**Pull request merged**